zypp/misc/CheckAccessDeleted.cc

```diff
@@ -241,59 +291,155 @@ namespace zypp
   } // namespace
   ///////////////////////////////////////////////////////////////////
 
-  CheckAccessDeleted::size_type CheckAccessDeleted::check( bool verbose_r )
+  CheckAccessDeleted::CheckAccessDeleted( bool doCheck_r )
+    : _pimpl(new Impl)
   {
-    _data.clear();
+    if ( doCheck_r ) check();
+  }
 
-    static const char* argv[] =
-    {
-      "lsof", "-n", "-FpcuLRftkn0", NULL
-    };
-    ExternalProgram prog( argv, ExternalProgram::Discard_Stderr );
+  CheckAccessDeleted::size_type CheckAccessDeleted::check( const Pathname &lsofOutput, bool verbose_r )
+  {
+    _pimpl->_verbose = verbose_r;
+    _pimpl->_fromLsofFileMode = true;
 
+    FILE *inFile = fopen( lsofOutput.c_str(), "r" );
+    if ( !inFile ) {
+      ZYPP_THROW( Exception( str::form("Opening input file %s failed.", lsofOutput.c_str()) ) );
```

Comment:
That's the _old_ printf style version. It's used a lot here and in zypper and AFAIK also in the libyiu-*-pkg packages. That's why I did not (yet) deprecate it.

In new code I try to switch to `str::Format` (based on [boost::format](http://www.boost.org/doc/libs/1_53_0/libs/format/)) and, especially if _translated strings_ are used,  to using _simple positional directives_.

    ZYPP_THROW( Exception( str::Format("Opening input file %1% failed.") % lsofOutput ) );

With translated strings we should use `str::FormatNAC` (*N*o *A*rgument *C*heck) to avoid lazy translators from breaking the code.

Reply:
Done